pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter.go

```diff
@@ -354,16 +357,24 @@ func (admitter *VMsAdmitter) authorizeVirtualMachineSpec(ar *admissionv1.Admissi
 			}
 		}
 
-		proxy := &authProxy{client: admitter.VirtClient, dataSourceInformer: admitter.DataSourceInformer, namespaceInformer: admitter.NamespaceInformer}
-		dv := &cdiv1.DataVolume{
-			ObjectMeta: dataVolume.ObjectMeta,
-			Spec:       dataVolume.Spec,
-		}
-		dv.Namespace = targetNamespace
-		allowed, message, err := admitter.cloneAuthFunc(dv, ar.Namespace, ar.Name, proxy, targetNamespace, serviceAccountName)
-		if err != nil && err != cdiv1.ErrNoTokenOkay {
+		allowed := true
+		message := ""
+		dv, err := storagetypes.GetDataVolumeFromCache(targetNamespace, dataVolume.Name, admitter.DataVolumeInformer.GetStore())
+		if err != nil {
 			return nil, err
 		}
+		if dv == nil {
+			dv = &cdiv1.DataVolume{
+				ObjectMeta: dataVolume.ObjectMeta,
+				Spec:       dataVolume.Spec,
+			}
+			dv.Namespace = targetNamespace
+			proxy := &authProxy{client: admitter.VirtClient, dataSourceInformer: admitter.DataSourceInformer, namespaceInformer: admitter.NamespaceInformer}
+			allowed, message, err = admitter.cloneAuthFunc(dv, ar.Namespace, ar.Name, proxy, targetNamespace, serviceAccountName)
+			if err != nil && err != cdiv1.ErrNoTokenOkay {
+				return nil, err
+			}
+		}
```

Comment:
Can't the auth call remain outside the if branch? I was thinking something like
```suggestion
		if dv == nil {
			dv = &cdiv1.DataVolume{
				ObjectMeta: dataVolume.ObjectMeta,
				Spec:       dataVolume.Spec,
			}
			dv.Namespace = targetNamespace
		}
```
And then initialization of allowed,msg could be dropped

Reply:
Not exactly sure what you are suggesting but I can clean that up